src/module/iotjs_module_fs.c

```diff
@@ -243,14 +243,14 @@ JHANDLER_FUNCTION(Read) {
 
   int fd = JHANDLER_GET_ARG(0, number);
   const iotjs_jval_t* jbuffer = JHANDLER_GET_ARG(1, object);
-  int offset = JHANDLER_GET_ARG(2, number);
-  int length = JHANDLER_GET_ARG(3, number);
+  size_t offset = JHANDLER_GET_ARG(2, number);
+  size_t length = JHANDLER_GET_ARG(3, number);
```

Comment:
The existing filter-logic below seems ok to check. In case of negative offset on unsigned type, integer underflow will occur. So, the condition, `(offset >= data_length)`, will be true.
```
if (offset >= data_length) {
    JHANDLER_THROW(RANGE, "offset out of bound");
    return;
}
```